Remora.Discord.Commands/Extensions/CommandTreeExtensions.cs

```diff
@@ -479,13 +480,33 @@ ILocalizationProvider localizationProvider
                 );
             }
 
-            if (command.CommandMethod.GetParameters().Length > 0)
+            var parameters = command.CommandMethod.GetParameters();
+            if (parameters.Length > 0)
             {
-                return new UnsupportedFeatureError
-                (
-                    "Context menu commands may not have parameters.",
-                    command
-                );
+                var expectedParameter = commandType switch
+                {
+                    ApplicationCommandType.Message => "message",
+                    ApplicationCommandType.User => "user",
+                    _ => null
```

Comment:
Throw here instead.

Reply:
Yeah throwing here would reduce the code right below this (since it will throw anyway since this sets it to null). It should help optimize because then it avoids an if which in this context would always be true when it returns null.